farmdata2_modules/fd2_tabs/resources/FarmOSAPI.spec.js

```diff
@@ -1,74 +1,215 @@
-        .then(() => {
-            cy.wait('@first').should(() => {
-                cy.wrap(testArray).should('have.length.gt',0)
-                firstCalls++
-            })
+            cy.wait(1)
+            .then(() => {
+                cy.wait('@first').should(() => {
+                    cy.wrap(testArray).should('have.length.gt',0)
+                    firstCalls++
+                })
 
-            cy.wait('@second').should(() => {
-                // Just by getting here we know the second page was requested.
+                cy.wait('@second').should(() => {
+                    // Just by getting here we know the second page was requested.
 
-                // Check that data made it into testArray.
-                // Note: depending on timing this may not run until after any
-                // subsequent calls.
-                cy.wrap(testArray).should('have.length.gt',100)
-                secondCalls++
-            })
+                    // Check that data made it into testArray.
+                    // Note: depending on timing this may not run until after any
+                    // subsequent calls.
+                    cy.wrap(testArray).should('have.length.gt',100)
+                    secondCalls++
+                })
 
-            cy.wait('@third').should(() => {
-                cy.wrap(testArray).should('have.length.gt',200)
-                thirdCalls++
-            })
+                cy.wait('@third').should(() => {
+                    cy.wrap(testArray).should('have.length.gt',200)
+                    thirdCalls++
+                })
 
-            getAllPages("/log?type=farm_seeding", testArray)
+                getAllPages("/log?type=farm_seeding", testArray)
 
+            })
+            .then(() => {
+                expect(firstCalls).to.equal(1)
+                expect(secondCalls).to.equal(1)
+                expect(thirdCalls).to.equal(1)
+            })
+        })
+    })
+    
+    context('test maping id to name functions', () => {
+        it('getIDToUserMap function returns a map', () => {
+            cy.wrap(getIDToUserMap()).as('map')
+            cy.get('@map').should((idToNameMap) => {
+                expect(idToNameMap).to.not.be.null
+                expect(idToNameMap).to.be.a('Map')
+            })
+        })
+        it('getIDToUserMap returns correct keys', () => {
+            cy.wrap(getIDToUserMap()).as('map')
+            cy.get('@map').should((idToNameMap) => {
+                expect(idToNameMap).to.have.all.keys(null, '6','8','7', '4', '9', '5', '1', '3', '155')
+            })
+        })
+        it('getIDToUserMap correct maps key to value', () => {
+            cy.wrap(getIDToUserMap()).as('map')
+            cy.get('@map').should((idToNameMap) => {
+                expect(idToNameMap.get(null)).to.equal('Anonymous')
+                expect(idToNameMap.get('5')).to.equal('worker1')
+            })
+        })
+        it('getIDToCropMap function returns a map', () => {
```

Comment:
Let's combine all of the tests for a given map into a single it.  So check that it is not null, a map, its size and a few of the specifics all in the same it.  That will reduce the number of tests and the time it takes to run them.